torch/csrc/cuda/GdsFile.cpp

```diff
@@ -0,0 +1,131 @@
+#include <pybind11/pybind11.h>
+#include <torch/csrc/utils/pybind.h>
+
+#if !defined(USE_ROCM) && defined(USE_CUFILE)
+#include <c10/cuda/CUDAGuard.h>
+
+#include <cuda_runtime.h>
+#include <cufile.h>
+
+// POSIX
+template <
+    class T,
+    typename std::enable_if<std::is_integral<T>::value, std::nullptr_t>::type =
+        nullptr>
+std::string cuGDSFileGetErrorString(T status) {
+  status = std::abs(status);
+  return IS_CUFILE_ERR(status) ? std::string(CUFILE_ERRSTR(status))
+                               : std::string(std::strerror(errno));
+}
+
+// CUfileError_t
+template <
+    class T,
+    typename std::enable_if<!std::is_integral<T>::value, std::nullptr_t>::type =
+        nullptr>
+std::string cuGDSFileGetErrorString(T status) {
+  std::string errStr = cuGDSFileGetErrorString(static_cast<int>(status.err));
+  if (IS_CUDA_ERR(status))
+    errStr.append(".").append(
+        cudaGetErrorString(static_cast<cudaError_t>(status.cu_err)));
+  return errStr;
+}
```

Comment:
Not sure what these are for?
But if these are just helpers, you can put them in an anonymous namespace to make sure they're not exposed

Reply:
Actually, inspecting this again, I think these two functions are right


`cuFileRead/Write` return ``size_t``, which has the following definition
<img width="1153" alt="Screenshot 2024-07-18 at 5 09 10 PM" src="https://github.com/user-attachments/assets/d5a089c0-9860-4198-9274-9d1572a8aba5">

On the other hand, `cuFile{Buf/Handle}{Register/Deregister}` return a struct `CUfileError_t` that contains `err` and `cuerr`, 

```

typedef struct CUfileError {
        CUfileOpError err; // cufile error
        enum CUresult cu_err; // for CUDA-specific errors
} CUfileError_t;

/**
 * error macros to inspect error status of type CUfileOpError
 */
 
#define IS_CUFILE_ERR(err) \
        (abs((err)) > CUFILEOP_BASE_ERR)
 
#define CUFILE_ERRSTR(err) \
        cufileop_status_error(static_cast<CUfileOpError>(abs((err))))
```

so keeping these two functions as is but adding comments to make this more clear